### DISCUSSION TITLE
When are enrichment tables reloaded?

### DISCUSSION THREAD

**Discussion by codersaur** [vectordotdev/vector#19782] [Q&A] [answered]
If a file referenced by an enrichment table changes, is the change automatically detected and the contents reloaded, or does vector need to be restarted? I ask as I don't see any log events emitted to indicate that changes to an enrichment table have been detected, despite using VECTOR_WATCH_CONFIG=true.
 
Also, does it make a difference if enrichment table files are stored in the VECTOR_CONFIG_DIR or elsewhere?

**Comment by jszwedko** [comment, marked as answer]
Vector needs to be restarted or reloaded to pick up changes to enrichment tables. Ideally `VECTOR_WATCH_CONFIG=true` would work too, but this is [an outstanding enhancement](https://github.com/vectordotdev/vector/issues/17283).

**Reply by coredump17** [reply]
is this still the case? Do you have any plans to allow an auto reload of the enrichment files?

**Reply by pront** [reply]
Yes, still the case as of today. The tracking issue for watching all config-adjacent files (enrichment tables, TLS certs, etc.) is #17283.
